### PR TITLE
Add toString builtin

### PR DESCRIPTION
**Pull request by sbarzowski**
+ Bugfix in string unparsing

### CODE REVIEW

**Comment by sbarzowski** [comment]
@sparkprime @lwander

File: interpreter.go [hunk 1]
```diff
@@ -460,7 +463,7 @@ func unparseString(v string) string {
 	for _, c := range v {
 		switch c {
 		case '"':
-			buf.WriteString("\n")
+			buf.WriteString("\\\"")
```
Comment (lwander) [inline interpreter.go]: Huh, was the `\n` a prior bug or is something special going on? Your changed string makes sense.

File: value.go [hunk 2]
```diff
@@ -61,11 +61,57 @@ func (v *valueBase) aValue() {}
...
+	}
+	for i := 0; i < length; i++ {
+		if a.value[i] != b.value[i] {
+			return a.value[i] < b.value[i]
+		}
+	}
+	return len(a.value) < len(b.value)
+}
+
+func (s *valueString) length() int {
```
Comment (lwander) [inline value.go]: Wasn't this added in a recent PR?

File: value.go [hunk 3]
```diff
@@ -61,11 +61,57 @@ func (v *valueBase) aValue() {}
...
 type valueString struct {
 	valueBase
-	value string
+	// We use rune slices instead of strings for quick indexing
+	value []rune
+}
+
+func (s *valueString) index(e *evaluator, index int) (value, error) {
+	runeIndex := 0
+	for _, r := range s.value {
```
Comment (lwander) [inline value.go]: Why not check if `index` is in the correct bounds, and if so, just index directly into `s.value`?

File: value.go (hunk 2 above)
Comment (lwander) [reply value.go]: Ah the PR wasn't merged yet :P sorry

File: value.go (hunk 3 above)
Comment (sbarzowski) [reply value.go]: Ah, when I was changing representation, I forgot to change that. Thanks (this will be changed in the other PR).

Comment (sbarzowski) [reply value.go]: done (in https://github.com/google/go-jsonnet/pull/34)

File: interpreter.go (hunk 1 above)
Comment (sbarzowski) [reply interpreter.go]: I think it was just a simple bug. There were no tests for that or anything. I don't see any reason to do that.

File: value.go (hunk 2 above)
Comment (sbarzowski) [reply value.go]: Actually it's turns out I don't need #34 here, so it's now based on master. Sorry for the confusion.

**Pull request merged**